Exporting files does not work

**Describe the bug**
A clear and concise description of what the bug is.

**To Reproduce**
1. Go to 'Settings'
2. Click on 'Export all'
3. Select folder (and press select)
4. Folder selection re-appears (or NullPointerException)

**Technical information**
 - OS: LineageOS
 - Version v3.3.1

Underlying issue seems to be:
```
01-20 03:52:41.423 E/AndroidRuntime(31594): FATAL EXCEPTION: main
01-20 03:52:41.423 E/AndroidRuntime(31594): Process: de.dennisguse.opentracks, PID: 31594
01-20 03:52:41.423 E/AndroidRuntime(31594): java.lang.RuntimeException: Unable to start activity ComponentInfo{de.dennisguse.opentracks/de.dennisguse.opentracks.io.file.exporter.ExportActivity}: androidx.fragment.app.Fragment$InstantiationException: Unable to instantiate fragment de.dennisguse.opentracks.fragments.FileTypeDialogFragment: could not find Fragment constructor
01-20 03:52:41.423 E/AndroidRuntime(31594): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2951)
01-20 03:52:41.423 E/AndroidRuntime(31594): 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3086)
01-20 03:52:41.423 E/AndroidRuntime(31594): 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78)
01-20 03:52:41.423 E/AndroidRuntime(31594): 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
01-20 03:52:41.423 E/AndroidRuntime(31594): 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
01-20 03:52:41.423 E/AndroidRuntime(31594): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1816)
01-20 03:52:41.423 E/AndroidRuntime(31594): 	at android.os.Handler.dispatchMessage(Handler.java:106)
01-20 03:52:41.423 E/AndroidRuntime(31594): 	at android.os.Looper.loop(Looper.java:193)
01-20 03:52:41.423 E/AndroidRuntime(31594): 	at android.app.ActivityThread.main(ActivityThread.java:6718)
01-20 03:52:41.423 E/AndroidRuntime(31594): 	at java.lang.reflect.Method.invoke(Native Method)
01-20 03:52:41.423 E/AndroidRuntime(31594): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
01-20 03:52:41.423 E/AndroidRuntime(31594): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
01-20 03:52:41.423 E/AndroidRunt
ime(31594): Caused by: androidx.fragment.app.Fragment$InstantiationException: Unable to instantiate fragment de.dennisguse.opentracks.fragments.FileTypeDialogFragment: could not find Fragment constructor
01-20 03:52:41.423 E/AndroidRuntime(31594): 	at androidx.fragment.app.Fragment.instantiate(Fragment.java:538)
01-20 03:52:41.423 E/AndroidRuntime(31594): 	at androidx.fragment.app.FragmentContainer.instantiate(FragmentContainer.java:57)
01-20 03:52:41.423 E/AndroidRuntime(31594): 	at androidx.fragment.app.FragmentManagerImpl$6.instantiate(FragmentManagerImpl.java:2851)
01-20 03:52:41.423 E/AndroidRuntime(31594): 	at androidx.fragment.app.FragmentState.instantiate(FragmentState.java:85)
01-20 03:52:41.423 E/AndroidRuntime(31594): 	at androidx.fragment.app.FragmentManagerImpl.restoreSaveState(FragmentManagerImpl.java:2488)
01-20 03:52:41.423 E/AndroidRuntime(31594): 	at androidx.fragment.app.FragmentController.restoreSaveState(FragmentController.java:195)
01-20 03:52:41.423 E/AndroidRuntime(31594): 	at androidx.fragment.app.FragmentActivity.onCreate(FragmentActivity.java:288)
01-20 03:52:41.423 E/AndroidRuntime(31594): 	at de.dennisguse.opentracks.io.file.exporter.ExportActivity.onCreate(ExportActivity.java:63)
01-20 03:52:41.423 E/AndroidRuntime(31594): 	at android.app.Activity.performCreate(Activity.java:7144)
01-20 03:52:41.423 E/AndroidRuntime(31594): 	at android.app.Activity.performCreate(Activity.java:7135)
01-20 03:52:41.423 E/AndroidRuntime(31594): 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1271)
01-20 03:52:41.423 E/AndroidRuntime(31594): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2931)
01-20 03:52:41.423 E/AndroidRuntime(31594): 	... 11 more
01-20 03:52:41.423 E/AndroidRuntime(31594): Caused by: java.lang.NoSuchMethodException: <init> []
01-20 03:52:41.423 E/AndroidRuntime(31594): 	at java.lang.Class.getConstructor0(Class.java:2327)
01-20 03:52:41.423 E/AndroidRuntime(31594): 	at java.lang.Class.getConstructor(Class.java:1725)
01-20 03:52:41.423 E/AndroidRuntime(31594): 	at androidx.fragment.app.Fragment.instantiate(Fragment.java:523)
01-20 03:52:41.423 E/AndroidRuntime(31594): 	... 22 more
```

--- a/src/androidTest/java/de/dennisguse/opentracks/fragments/ChartFragmentTest.java
+++ b/src/androidTest/java/de/dennisguse/opentracks/fragments/ChartFragmentTest.java
@@ -61,8 +61,7 @@ public class ChartFragmentTest {
     @Before
     public void setUp() {
         boolean chartByDistance = false;
-        chartFragment = new ChartFragment(chartByDistance) {
-        };
+        chartFragment = (ChartFragment) ChartFragment.newInstance(chartByDistance);
         chartFragment.setChartView(new ChartView(ApplicationProvider.getApplicationContext(), chartByDistance));
         chartFragment.setTripStatisticsUpdater(TrackStubUtils.INITIAL_TIME);
     }

--- a/src/main/java/de/dennisguse/opentracks/TrackDetailActivity.java
+++ b/src/main/java/de/dennisguse/opentracks/TrackDetailActivity.java
@@ -34,7 +34,6 @@ import android.view.WindowManager;
 import android.widget.Toast;
 
 import androidx.annotation.NonNull;
-import androidx.annotation.Nullable;
 import androidx.core.app.ActivityCompat;
 import androidx.fragment.app.Fragment;
 import androidx.fragment.app.FragmentPagerAdapter;
@@ -46,8 +45,7 @@ import de.dennisguse.opentracks.content.TrackDataHub;
 import de.dennisguse.opentracks.content.data.Track;
 import de.dennisguse.opentracks.content.data.Waypoint;
 import de.dennisguse.opentracks.content.provider.ContentProviderUtils;
-import de.dennisguse.opentracks.fragments.ChartDistanceFragment;
-import de.dennisguse.opentracks.fragments.ChartTimeFragment;
+import de.dennisguse.opentracks.fragments.ChartFragment;
 import de.dennisguse.opentracks.fragments.ChooseActivityTypeDialogFragment;
 import de.dennisguse.opentracks.fragments.ConfirmDeleteDialogFragment;
 import de.dennisguse.opentracks.fragments.StatsFragment;
@@ -206,17 +204,15 @@ public class TrackDetailActivity extends AbstractListActivity implements ChooseA
             @Override
             public Fragment getItem(int position) {
                 switch (position) {
-                    case 0:
-                        return new StatsFragment();
                     case 1:
-                        return new ChartTimeFragment();
+                        return ChartFragment.newInstance(false);
                     case 2:
-                        return new ChartDistanceFragment();
+                        return ChartFragment.newInstance(true);
+                    default: //0
+                        return new StatsFragment();
                 }
-                return null;
             }
 
-            @Nullable
             @Override
             public CharSequence getPageTitle(int position) {
                 switch (position) {

--- a/src/main/java/de/dennisguse/opentracks/fragments/ChartDistanceFragment.java
+++ b/src/main/java/de/dennisguse/opentracks/fragments/ChartDistanceFragment.java
@@ -1,8 +1,0 @@
-package de.dennisguse.opentracks.fragments;
-
-public class ChartDistanceFragment extends ChartFragment {
-
-    public ChartDistanceFragment() {
-        super(true);
-    }
-}

--- a/src/main/java/de/dennisguse/opentracks/fragments/ChartFragment.java
+++ b/src/main/java/de/dennisguse/opentracks/fragments/ChartFragment.java
@@ -25,7 +25,6 @@ import android.view.ViewGroup;
 import android.view.ViewGroup.LayoutParams;
 
 import androidx.annotation.NonNull;
-import androidx.annotation.Nullable;
 import androidx.annotation.VisibleForTesting;
 import androidx.fragment.app.Fragment;
 import androidx.fragment.app.FragmentActivity;
@@ -56,9 +55,18 @@ import de.dennisguse.opentracks.util.UnitConversions;
  * @author Sandor Dornbush
  * @author Rodrigo Damazio
  */
-public abstract class ChartFragment extends Fragment implements TrackDataListener {
+public class ChartFragment extends Fragment implements TrackDataListener {
 
-    private static final String STATE_CHART_VIEW_BY_DISTANCE_KEY = "chartViewByDistance";
+    private static final String KEY_CHART_VIEW_BY_DISTANCE_KEY = "chartViewByDistance";
+
+    public static Fragment newInstance(boolean chartByDistance) {
+        Bundle bundle = new Bundle();
+        bundle.putBoolean(KEY_CHART_VIEW_BY_DISTANCE_KEY, chartByDistance);
+
+        ChartFragment chartFragment = new ChartFragment();
+        chartFragment.setArguments(bundle);
+        return chartFragment;
+    }
 
     private final List<double[]> pendingPoints = new ArrayList<>();
 
@@ -139,13 +147,12 @@ public abstract class ChartFragment extends Fragment implements TrackDataListene
         }
     };
 
-    public ChartFragment(boolean chartByDistance) {
-        this.chartByDistance = chartByDistance;
-    }
 
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+
+        chartByDistance = getArguments().getBoolean(KEY_CHART_VIEW_BY_DISTANCE_KEY, chartByDistance);
 
         recordingDistanceInterval = PreferencesUtils.getRecordingDistanceIntervalDefault(getContext());
 
@@ -188,20 +195,6 @@ public abstract class ChartFragment extends Fragment implements TrackDataListene
         super.onStop();
         ViewGroup layout = getView().findViewById(R.id.chart_view_layout);
         layout.removeView(chartView);
-    }
-
-    @Override
-    public void onSaveInstanceState(@NonNull Bundle outState) {
-        super.onSaveInstanceState(outState);
-        outState.putBoolean(STATE_CHART_VIEW_BY_DISTANCE_KEY, chartByDistance);
-    }
-
-    @Override
-    public void onViewStateRestored(@Nullable Bundle savedInstanceState) {
-        super.onViewStateRestored(savedInstanceState);
-        if (savedInstanceState != null) {
-            chartByDistance = savedInstanceState.getBoolean(STATE_CHART_VIEW_BY_DISTANCE_KEY);
-        }
     }
 
     @Override

--- a/src/main/java/de/dennisguse/opentracks/fragments/ChartTimeFragment.java
+++ b/src/main/java/de/dennisguse/opentracks/fragments/ChartTimeFragment.java
@@ -1,8 +1,0 @@
-package de.dennisguse.opentracks.fragments;
-
-public class ChartTimeFragment extends ChartFragment {
-
-    public ChartTimeFragment() {
-        super(false);
-    }
-}

--- a/src/main/java/de/dennisguse/opentracks/fragments/ConfirmDeleteDialogFragment.java
+++ b/src/main/java/de/dennisguse/opentracks/fragments/ConfirmDeleteDialogFragment.java
@@ -36,12 +36,9 @@ import de.dennisguse.opentracks.util.DialogUtils;
 public class ConfirmDeleteDialogFragment extends DialogFragment {
 
     private static final String CONFIRM_DELETE_DIALOG_TAG = "confirmDeleteDialog";
-    private ConfirmDeleteCaller caller;
-    private final long[] trackIds;
+    private static final String KEY_TRACK_IDS = "trackIds";
 
-    public ConfirmDeleteDialogFragment(long[] trackIds) {
-        this.trackIds = trackIds;
-    }
+    private ConfirmDeleteCaller caller;
 
     /**
      * Create a new instance.
@@ -49,7 +46,12 @@ public class ConfirmDeleteDialogFragment extends DialogFragment {
      * @param trackIds list of track ids to delete. To delete all, set to size 1 with trackIds[0] == -1L
      */
     public static void showDialog(FragmentManager fragmentManager, long[] trackIds) {
-        new ConfirmDeleteDialogFragment(trackIds).show(fragmentManager, CONFIRM_DELETE_DIALOG_TAG);
+        Bundle bundle = new Bundle();
+        bundle.putLongArray(KEY_TRACK_IDS, trackIds);
+
+        ConfirmDeleteDialogFragment deleteTrackDialogFragment = new ConfirmDeleteDialogFragment();
+        deleteTrackDialogFragment.setArguments(bundle);
+        deleteTrackDialogFragment.show(fragmentManager, CONFIRM_DELETE_DIALOG_TAG);
     }
 
     @Override
@@ -65,6 +67,8 @@ public class ConfirmDeleteDialogFragment extends DialogFragment {
     @Override
     @NonNull
     public Dialog onCreateDialog(Bundle savedInstanceState) {
+        final long[] trackIds = getArguments().getLongArray(KEY_TRACK_IDS);
+
         int titleId;
         int messageId;
         titleId = trackIds.length > 1 ? R.string.generic_delete_selected_confirm_title : R.string.track_delete_one_confirm_title;

--- a/src/main/java/de/dennisguse/opentracks/fragments/DeleteMarkerDialogFragment.java
+++ b/src/main/java/de/dennisguse/opentracks/fragments/DeleteMarkerDialogFragment.java
@@ -38,16 +38,17 @@ import de.dennisguse.opentracks.util.DialogUtils;
 public class DeleteMarkerDialogFragment extends DialogFragment {
 
     private static final String DELETE_MARKER_DIALOG_TAG = "deleteMarkerDialog";
-    private final long[] markerIds;
+    private static final String KEY_MARKER_IDS = "markerIds";
 
     private DeleteMarkerCaller caller;
 
-    public DeleteMarkerDialogFragment(long[] markerIds) {
-        this.markerIds = markerIds;
-    }
-
     public static void showDialog(FragmentManager fragmentManager, long[] markerIds) {
-        new DeleteMarkerDialogFragment(markerIds).show(fragmentManager, DELETE_MARKER_DIALOG_TAG);
+        Bundle bundle = new Bundle();
+        bundle.putLongArray(KEY_MARKER_IDS, markerIds);
+
+        DeleteMarkerDialogFragment deleteMarkerDialogFragment = new DeleteMarkerDialogFragment();
+        deleteMarkerDialogFragment.setArguments(bundle);
+        deleteMarkerDialogFragment.show(fragmentManager, DELETE_MARKER_DIALOG_TAG);
     }
 
     @Override
@@ -63,6 +64,8 @@ public class DeleteMarkerDialogFragment extends DialogFragment {
     @Override
     @NonNull
     public Dialog onCreateDialog(Bundle savedInstanceState) {
+        final long[] markerIds = getArguments().getLongArray(KEY_MARKER_IDS);
+
         final FragmentActivity fragmentActivity = getActivity();
         int titleId;
         int messageId;

--- a/src/main/java/de/dennisguse/opentracks/fragments/FileTypeDialogFragment.java
+++ b/src/main/java/de/dennisguse/opentracks/fragments/FileTypeDialogFragment.java
@@ -40,18 +40,19 @@ import de.dennisguse.opentracks.io.file.TrackFileFormat;
 public class FileTypeDialogFragment extends DialogFragment {
 
     private static final String FILE_TYPE_DIALOG_TAG = "fileType";
-    private final int titleId;
+    private static final String KEY_TITLE_ID = "titleId";
+    private static final String KEY_OPTION_ID = "optionId";
 
     private FileTypeCaller caller;
-    private final int optionId;
-
-    public FileTypeDialogFragment(int titleId, int optionId) {
-        this.titleId = titleId;
-        this.optionId = optionId;
-    }
 
     public static void showDialog(FragmentManager fragmentManager, int titleId, int optionId) {
-        new FileTypeDialogFragment(titleId, optionId).show(fragmentManager, FILE_TYPE_DIALOG_TAG);
+        Bundle bundle = new Bundle();
+        bundle.putInt(KEY_TITLE_ID, titleId);
+        bundle.putInt(KEY_OPTION_ID, optionId);
+
+        FileTypeDialogFragment fileTypeDialogFragment = new FileTypeDialogFragment();
+        fileTypeDialogFragment.setArguments(bundle);
+        fileTypeDialogFragment.show(fragmentManager, FILE_TYPE_DIALOG_TAG);
     }
 
     @Override
@@ -67,6 +68,9 @@ public class FileTypeDialogFragment extends DialogFragment {
     @Override
     @NonNull
     public Dialog onCreateDialog(Bundle savedInstanceState) {
+        int optionId = getArguments().getInt(KEY_OPTION_ID);
+        final int titleId = getArguments().getInt(KEY_TITLE_ID);
+
         final TrackFileFormat[] trackFileFormats = {TrackFileFormat.KMZ_WITH_TRACKDETAIL_AND_SENSORDATA_AND_PICTURES, TrackFileFormat.KMZ_WITH_TRACKDETAIL_AND_SENSORDATA, TrackFileFormat.KML_WITH_TRACKDETAIL_AND_SENSORDATA, TrackFileFormat.GPX};
         String[] choices = new String[trackFileFormats.length];
         for (int i = 0; i < choices.length; i++) {

--- a/src/main/java/de/dennisguse/opentracks/fragments/MarkerDetailFragment.java
+++ b/src/main/java/de/dennisguse/opentracks/fragments/MarkerDetailFragment.java
@@ -57,7 +57,9 @@ public class MarkerDetailFragment extends Fragment {
 
     private static final String TAG = MarkerDetailFragment.class.getSimpleName();
     private static final String KEY_MARKER_ID = "markerId";
+
     private static final long HIDE_TEXT_DELAY = 4 * UnitConversions.ONE_SECOND_MS;
+
     private ContentProviderUtils contentProviderUtils;
     private Handler handler;
     private ImageView photoView;

--- a/src/main/java/de/dennisguse/opentracks/io/file/exporter/ExportAsyncTask.java
+++ b/src/main/java/de/dennisguse/opentracks/io/file/exporter/ExportAsyncTask.java
@@ -32,7 +32,6 @@ import de.dennisguse.opentracks.content.data.Track;
 import de.dennisguse.opentracks.content.data.TracksColumns;
 import de.dennisguse.opentracks.content.provider.ContentProviderUtils;
 import de.dennisguse.opentracks.io.file.TrackFileFormat;
-import de.dennisguse.opentracks.util.PreferencesUtils;
 import de.dennisguse.opentracks.util.SystemUtils;
 
 /**
@@ -98,17 +97,12 @@ public class ExportAsyncTask extends AsyncTask<Void, Integer, Boolean> {
     protected Boolean doInBackground(Void... params) {
         try {
             Thread.currentThread().setPriority(Thread.MAX_PRIORITY);
-            boolean isRecording = PreferencesUtils.isRecording(exportActivity);
-            boolean isPaused = PreferencesUtils.isRecordingTrackPaused(exportActivity);
-            // Get the wake lock if not recording or paused
-            if (!isRecording || isPaused) {
-                wakeLock = SystemUtils.acquireWakeLock(exportActivity, wakeLock);
-            }
+
+            wakeLock = SystemUtils.acquireWakeLock(exportActivity, wakeLock);
+
             return exportAllTracks();
         } finally {
-            if (wakeLock != null && wakeLock.isHeld()) {
-                wakeLock.release();
-            }
+            wakeLock = SystemUtils.releaseWakeLock(wakeLock);
         }
     }
 

